cylc/flow/task_pool.py

```diff
@@ -47,26 +40,111 @@
 from cylc.flow.task_proxy import TaskProxy
 from cylc.flow.task_state import (
     TASK_STATUSES_ACTIVE, TASK_STATUSES_FAILURE, TASK_STATUSES_NOT_STALLED,
-    TASK_STATUS_WAITING, TASK_STATUS_EXPIRED,
+    TASK_STATUSES_SUCCESS, TASK_STATUS_WAITING, TASK_STATUS_EXPIRED,
     TASK_STATUS_QUEUED, TASK_STATUS_READY, TASK_STATUS_SUBMITTED,
     TASK_STATUS_SUBMIT_FAILED, TASK_STATUS_SUBMIT_RETRYING,
     TASK_STATUS_RUNNING, TASK_STATUS_SUCCEEDED, TASK_STATUS_FAILED,
-    TASK_STATUS_RETRYING)
+    TASK_STATUS_RETRYING, TASK_OUTPUT_SUCCEEDED, TASK_OUTPUT_FAILED,
+    TASK_OUTPUT_EXPIRED)
 from cylc.flow.wallclock import get_current_time_string
 
 
+class FlowLabelMgr(object):
+    """
+    Manage flow labels consisting of a string of one or more letters [a-zA-Z].
+
+    Flow labels are task attributes representing the flow the task belongs to,
+    passed down to spawned children. If a new flow is started, a new single
+    character label is chosen randomly. This allows for 52 simultaneous flows
+    (which should be more than enough) with labels that are easy to work with.
```

Comment:
:rofl: it would be trivial to extend this to an arbitrary number of concurrent flows (e.g. UUID labels, with sets of multiple labels for merging) however not quite so easy to make the resulting labels easy for users to refer to (athough associated metadata might help).